core/src/main/java/com/sequenceiq/cloudbreak/converter/v4/stacks/StackV4RequestToStackConverter.java

```diff
@@ -380,15 +380,13 @@ private void validateStackAuthentication(StackV4Request source) {
     }
 
     private boolean shouldCreateInternalKnoxLoadBalancer(StackType type, EnvironmentNetworkResponse network) {
-        // TODO expand this to data hubs
-        return StackType.DATALAKE.equals(type) &&
+        return (StackType.DATALAKE.equals(type) || StackType.WORKLOAD.equals(type)) &&
             (entitlementService.datalakeLoadBalancerEnabled(ThreadBasedUserCrnProvider.getAccountId())
```

Comment:
`entitlementService.datalakeLoadBalancerEnabled()` should only be considered if the stack is of type `DATALAKE`. We don't want the data hub logic to trigger except for semi-private networks. So it'd be something like
```
return (StackType.DATALAKE.equals(type) && entitlementService.datalakeLoadBalancerEnabled(ThreadBasedUserCrnProvider.getAccountId()))
            || ((StackType.DATALAKE.equals(type) || StackType.WORKLOAD.equals(type)) && isEndpointGatewayEnabled(network));
```

Reply:
This logic was something like that before one of the rebases. Let me see do one more attempt.